cypress/plugins/index.ts

```diff
@@ -36,6 +38,18 @@ const configurePlugins: Cypress.PluginConfig = (on) => {
       webpackOptions,
     })
   );
+  on("after:spec", (spec, results) => {
+    if (results?.video) {
+      // Do we have failures for any retry attempts?
+      const failures = _.some(results.tests, (test: any) => {
+        return _.some(test.attempts, { state: "failed" });
+      });
+      if (!failures) {
+        // delete the video if the spec passed and no tests retried
+        return del(results.video);
+      }
+    }
```

Comment:
I think it should be possible to accomplish the same without the need for external libraries.
```suggestion
    if (!results.video) {
      return;
    }

    // Do we have failures for any retry attempts?
    const failures = results.tests.some(({ attempts }) =>
      attempts.some(({ state }) => state === "failed")
    );

    // delete the video if the spec passed and no tests retried
    if (!failures) {
      fs.rmSync(results.video);
    }
```